Add util function for pretty printing of output diffs

AdrianLundell opened the pull request:
The function is added as a callback to run_method_and_compare_outputs to open up for other types  of analyze methods in the future.

AdrianLundell committed:
Add util function for pretty printing of output diffs

Change-Id: I93416fe3f2c175050b84a16004bf32849bafd1fe

AdrianLundell commented:
Unrelated CI failure in 
- backends/xnnpack/test/ops/test_add.py::TestAdd::test_qs8_add:  https://github.com/pytorch/executorch/actions/runs/12300250961/job/34327986575?pr=7302#step:14:7721
- examples/models/llama3_2_vision/text_decoder/test/test_text_decoder.py: https://github.com/pytorch/executorch/actions/runs/12300250961/job/34327988517?pr=7302#step:9:16682

freddan80 committed:
Update analyze_output_utils.py

freddan80 committed:
Update analyze_output_utils.py

freddan80 commented on backends/arm/test/tester/analyze_output_utils.py:
```diff
@@ -0,0 +1,216 @@
+
+import torch
+
+logger = logging.getLogger(__name__)
```
```suggestion
logger = logging.getLogger(__name__) # Test suggest change
```

freddan80 replied:
Ignore. Just a test

AdrianLundell committed:
Merge branch 'pytorch:main' into dev/analyze_output_utils

AdrianLundell committed:
Move callbacks into _compare_outputs

This is to make run_method_and_compare_outputs less complex since lintrunner was complaining.
Additionally moves out previous info dumps in compare_output into a new callback function to handle all error handling in the same way.

Change-Id: If7c7cfd515d4870c018a34785dd80be15d4fbcef

AdrianLundell committed:
Merge branch 'main' of https://github.com/pytorch/executorch into dev/analyze_output_utils

AdrianLundell commented:
Unrelated CI error when running 

```
curl -Lo /tmp/v2.25.0.24.07.28.zip https://softwarecenter.qualcomm.com/api/download/software/qualcomm_neural_processing_sdk/v2.25.0.240728.zip 
...
2024-12-18T08:56:22.4369579Z curl: (92) HTTP/2 stream 0 was not closed cleanly: INTERNAL_ERROR (err 2)
2024-12-18T08:56:22.4370216Z     raise RuntimeError(f"Command {cmd} failed with exit code {exit_code}")
```

AdrianLundell committed:
Merge branch 'pytorch:main' into dev/analyze_output_utils
